File: backend/src/main/java/com/example/backend/entity/Review.java
package com.example.backend.entity;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Review {
    private double rating;
    private String userUID;
    private String comment;

    private String username;
    private String userImageURL;

    private int recipeID;
    private Boolean isExternal;

    public ReviewDTO convertToDTO() {
        ReviewDTO dto = new ReviewDTO();
        dto.setComment(this.comment);
        dto.setRating(this.rating);
        dto.setUserUID(this.userUID);
        return dto;
    }
}

File: frontend/lib/recipe/models/review.dart
import 'package:frontend/recipe/models/review_DTO.dart';

class Review {
  Review(
      {required this.rating,
      required this.userUID,
      required this.username,
      required this.comment,
      required this.userImageURL});
  final double rating;
  final String userUID;
  final String username;
  final String comment;
  final String userImageURL;

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
        rating: json['rating'],
        userUID: json['userUID'],
        username: json['username'],
        comment: json['comment'],
        userImageURL: json['userImageURL']);
  }

  ReviewDTO toDTO(int recipeID, bool isExternal) {
    return ReviewDTO(
        rating: rating,
        userUID: userUID,
        comment: comment,
        recipeID: recipeID,
        isExternal: isExternal);
  }
}
